sql/src/test/java/io/crate/integrationtests/IngestionServiceIntegrationTest.java

```diff
@@ -0,0 +1,228 @@
+                    implementation.ingestData(DATA_TOPIC, nextValue);
+                    lastProducedData.set(nextValue);
+                } catch (InterruptedException e) {
+                    // will retry if interrupted
+                } finally {
+                    dataFlowSemaphore.release();
+                }
+                dataFlowingLatch.countDown();
+            }
+        }, "DataIngestionThread");
+        dataIngestionThread.start();
+    }
+
+    @After
+    public void dropIngestRuleAndStopProducingData() throws InterruptedException {
+        execute("drop ingest rule if exists " + UNDER_TEST_INGEST_RULE_NAME);
+        resumeDataIngestion();
+        produceData = false;
+        dataIngestionThread.join();
+    }
+
+    @Test
+    public void testIngestionImplementationIngestsData() throws Exception {
+        implementation.registerIngestionImplementation();
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+        pauseDataIngestion();
+
+        execute("select * from ingest_data_raw order by data desc");
+        assertThat(response.rows()[0][0], is(lastProducedData.get()));
+    }
+
+    @Test
+    public void testCreateRuleOnDifferentTopicThanIncomingData() throws Exception {
+        implementation.registerIngestionImplementation();
+        execute("create table other_topic_raw_table (data int)");
+        execute(String.format(Locale.ENGLISH,
+            "create ingest rule rule_for_other_topic on %s where topic = '%s' into %s",
+            INGESTION_SOURCE_ID, "other_topic", "other_topic_raw_table")
+        );
+
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+        execute("select * from other_topic_raw_table order by data desc");
+        assertThat(response.rowCount(), is(0L));
+    }
+
+    @Test
+    public void testCreateMultipleRulesForTheSameTargetTable() throws Exception {
+        implementation.registerIngestionImplementation();
+        execute(String.format(Locale.ENGLISH,
+            "create ingest rule another_topic_rule on %s where topic = '%s' into %s",
+            INGESTION_SOURCE_ID, DATA_TOPIC, TARGET_TABLE)
+        );
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+        pauseDataIngestion();
+
+        // last produced data should've been ingested twice in the target table
+        execute("select * from ingest_data_raw order by data desc limit 2");
+        assertThat(response.rows()[0][0], is(lastProducedData.get()));
+        assertThat(response.rows()[1][0], is(lastProducedData.get()));
+    }
+
+    @Test
+    public void testDropIngestRuleWillCauseImplementationToStopIngesting() throws Exception {
+        implementation.registerIngestionImplementation();
+        dataFlowingLatch.await(5, TimeUnit.SECONDS);
+
+        pauseDataIngestion();
+        execute("drop ingest rule " + UNDER_TEST_INGEST_RULE_NAME);
+        int expectedLastIngestedValue = lastProducedData.get();
+        resumeDataIngestion();
```

Comment:
afaik ingestion must be paused to get the accurate last produced/ingested data. can you pls add a comment here?